docs/developing-contracts/testing.md

```diff
@@ -0,0 +1,298 @@
+
+### Import `gtest` lib
+
+In order to use the `gtest` library, it must be imported into your `Cargo.toml` file in the `[dev-dependencies]` block in order to fetch and compile it for tests only
+
+```rust
+[package]
+name = "first-gear-app"
+version = "0.1.0"
+authors = ["Your Name"]
+edition = "2021"
+license = "GPL-3.0"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+gstd = { git = "https://github.com/gear-tech/gear.git", features = ["debug"] }
+
+[dev-dependencies]
+gtest = { git = "https://github.com/gear-tech/gear.git" }
+
+[profile.release]
+lto = true
+opt-level = 's'
+```
+
+### Possible issues
+
+When writing tests as integration tests, it is not possible to import public structures from within the program itself, because gear contracts must have `crate-type=["cdylib"]` struct and nothing can be imported from them.
+
+Solution options:
+* (Recommended) Create a sub-crate library with IO structures and import it into `Cargo.toml`. In this case, structures will be available everywhere.
+* Redefine the same structures in the test file (copy-paste), because with codec they will produce the same bytes.
+
+## `gtest` capabilities
+
+The example provided for [PING-PONG](./developing-contracts/examples/ping.md)  program.
+
+```rust
+use gtest::{Log, Program, System};
+
+#[test]
+fn basics() {
+    // Initialization of the common environment for running smart contacts.
+    //
+    // This emulates node's and chain behavior.
+    //
+    // By default, sets:
+    // - current block equal 0
+    // - current timestamp equals UNIX timestamp of your system.
+    // - minimal message id equal 0x010000..
+    // - minimal program id equal 0x010000..
+    let sys = System::new();
+
+    // You may control time in the system by spending blocks.
+    //
+    // It adds the amount of blocks passed as argument to the current block of the system.
+    // Same for the timestamp. Note, that for now 1 block in Gear network is 1 sec duration.
+    sys.spend_blocks(150);
+
+    // Initialization of styled `env_logger` to print logs (only from `gwasm` by default) into stdout.
+    //
+    // To specify printed logs, set the env variable `RUST_LOG`:
+    // `RUST_LOG="target_1=logging_level,target_2=logging_level" cargo test`
+    //
+    // Gear smart contracts use `gwasm` target with `debug` logging level
+    sys.init_logger();
+
+    // Initialization of program structure from file.
+    //
+    // Takes as arguments reference to the related `System` and the path to wasm binary relatively 
+    // the root of the crate where the test was written.
+    //
+    // Sets free program id from the related `System` to this program. For this case it equals 0x010000..
+    // Next program initialized without id specification will have id 0x020000.. and so on.
+    let ping_pong = Program::from_file(
+        &sys,
+        "../target/wasm32-unknown-unknown/release/demo_ping.wasm",
```

Comment:
Replace everywhere "../target" with "./target", my fault due to workspace in apps repo

Reply:
done